### ISSUE TITLE
nothing happens on "New Comment" - JS seems to not be executing

### ISSUE THREAD

**Issue by cpg** [lml/commontator#86]
I tried this gem and all was ok, up to the point of using the app (rails 4.2.5) with a browser. Tried FF and Chrome (both latest on mac os x).

Looks like the JS coming back from the AJAX calls is not executed. JS is enabled, needless to say.

The call to the server to open up a new comment `Commontator::CommentsController#new` is done ok, comes back having rendered the gem's `_form.html.erb` and `new.js.erb` well, but nothing happens.

No errors in the browser console, no action on the screen. The form does not show up on the markup (unlike in #47).

I tried disabling turbolinks (in development) without much luck. I then set `new_comment_style = :t` and I could see the form, then post the first comment. Again, no action when posting, even though all these things in the gem are rendered properly. Refreshing the page does render the comments as expecting.

In the case of the "New Comment" i see this but nothing inside the div after pressing the link.
```
<div id="thread_1_new_comment_form_div" class="thread_new_comment_form commontator_hidden">
 </div>
```
If I read the `comments/new.js.erb` correctly, that div should be populated, but I do not see where this is failing.

Any ideas?

Thanks in advance and great job on the gem.

**Comment by Dantemss** [comment]
Check if `jquery-ujs` is in your `application.rb`

**Comment by cpg** [comment]
if you mean application.js, yes, I have it I forgot to mention that.

```
//= require jquery_ujs
```

**Comment by cpg** [comment]
I should also mention that I am using bootstrap.

**Comment by Dantemss** [comment]
I am a little baffled myself... is it possible to provide a minimal app that reproduces the issue?

**Comment by Dantemss** [comment]
Or rather, before you do that, does this happen on all Commontator JS (for example, closing threads etc) or just on new/create?

**Comment by cpg** [comment]
Deleting a comment also does this, deletes the comment but makes no UI changes and the web console does not show anything at all, so I do not have a lot to debug with. Other things I have 

In case it helps, if I look at the event hooked to the "New Comment" link, I see this code (coming from jquery_ujs):
```
function(e) {
  var link = $(this),
    method = link.data('method'),
    data = link.data('params'),
    metaClick = e.metaKey || e.ctrlKey;
  if (!rails.allowAction(link)) return rails.stopEverything(e);

  if (!metaClick && link.is(rails.linkDisableSelector)) rails.disableElement(link);

  if (rails.isRemote(link)) {
    if (metaClick && (!method || method === 'GET') && !data) {
      return true;
    }

    var handleRemote = rails.handleRemote(link);
    // response from rails.handleRemote() will either be false or a deferred object promise.
    if (handleRemote === false) {
      rails.enableElement(link);
    } else {
      handleRemote.fail(function() {
        rails.enableElement(link);
      });
    }
    return false;

  } else if (method) {
    rails.handleMethod(link);
    return false;
  }
}
```

**Comment by cpg** [comment]
Oh, reviewing things for possible things that could be interfering, I found a big clue. I have this code in the app. When I remove it things work as expected!
```
$.ajaxSetup({
	global: true,
	dataType: 'json',
	contentType: 'application/x-www-form-urlencoded; charset=utf-8'
});
```

I had this there for a while though I do not remember well why -- I think it was related to turbolinks interactions causing trouble with d3.js or something else. Doing a search for code similar to it, I see [in this post](http://stackoverflow.com/questions/6996993/jquerys-ajaxsetup-i-would-like-to-add-default-data-for-get-requests-only) that it may have been "to ensure cache-miss when a new revision is deployed and the frontend ask for html templates or json-data from the backend."

But I confess that I do not quite understand the core reasons we added that code, it was just to "fix" some bad interaction.

I will test the site without this code. Thanks for your patience.
